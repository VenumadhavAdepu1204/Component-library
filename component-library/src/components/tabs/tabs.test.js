import React from 'react'
import Tab from './index'
import {render, fireEvent, cleanup} from '@testing-library/react'
describe('testing tab component', () => {
  afterEach(cleanup)
  test('renders properly', () => {
    const {container} = render(<Tab><div label='one'>one</div>
      <div label='two'>two</div>
      <div label='three'>three</div>
    </Tab>)
    expect(container.firstChild).toMatchSnapshot()
  })
  test('renders properly with border and background props', () => {
    const {container} = render(<Tab clickBorder='blue' hoverBackground='black' selectedLabel='one'>
      <div label='one'>one</div>
      <div label='two'>two</div>
      <div label='three'>three</div>
    </Tab>)
    expect(container.firstChild).toMatchSnapshot()
  })
  test('click of tab should render the child ', () => {
    const {container, getByTestId} = render(<Tab><div label='one'>one</div>
      <div label='two'>two</div>
      <div label='three'>three</div>
    </Tab>)
    fireEvent.click(getByTestId('one'))
    expect(container.children.length).not.toBe(0)
  })
  test('keypress of enter,space should render the child', () => {
    const {container, getByTestId} = render(<Tab><div label='one'>one</div>
      <div label='two'>two</div>
      <div label='three'>three</div>
    </Tab>)
    fireEvent.keyUp(getByTestId('one'), {key: 'Enter', code: 13})
    expect(container.children.length).not.toBe(0)
  })
  test('keypress of enter,space should render the child', () => {
    const {container, getByTestId} = render(<Tab><div label='one'>one</div>
      <div label='two'>two</div>
      <div label='three'>three</div>
    </Tab>)
    fireEvent.keyUp(getByTestId('one'), {key: 'Enter', code: 13, keyCode: 13})
    fireEvent.keyUp(getByTestId('one'), {key: 'Space', code: 32, keyCode: 32})
    expect(container.children.length).not.toBe(0)
  })
  test('keypress of arrows should move the focus to tabs', () => {
    const {getByTestId} = render(<Tab><div label='one'>one</div>
      <div label='two'>two</div>
      <div label='three'>three</div>
    </Tab>)
    fireEvent.keyUp(getByTestId('one'), {key: 'Arrow', code: 37, keyCode: 37})
    fireEvent.keyUp(getByTestId('one'), {key: 'Arrow', code: 39, keyCode: 39})
    fireEvent.keyUp(getByTestId('one'), {key: 'Arrow', code: 38, keyCode: 38})
    fireEvent.keyUp(getByTestId('one'), {key: 'Arrow', code: 40, keyCode: 40})
  })
})
