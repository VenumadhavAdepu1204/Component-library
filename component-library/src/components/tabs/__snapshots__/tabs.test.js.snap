// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing tab component renders properly 1`] = `
<div
  class="sc-bdVaJa jboVGq"
>
  <ul
    class="sc-bwzfXH fORQjs"
  >
    <li
      class="sc-ifAKCX jwJMkm"
      data-testid="one"
      tabindex="0"
    >
      <div
        class="sc-bxivhb eBNJKg"
      >
        one
         
      </div>
    </li>
    <li
      class="sc-ifAKCX jwJMkm"
      data-testid="two"
      tabindex="0"
    >
      <div
        class="sc-bxivhb eBNJKg"
      >
        two
         
      </div>
    </li>
    <li
      class="sc-ifAKCX jwJMkm"
      data-testid="three"
      tabindex="0"
    >
      <div
        class="sc-bxivhb eBNJKg"
      >
        three
         
      </div>
    </li>
  </ul>
</div>
`;

exports[`testing tab component renders properly with border and background props 1`] = `
<div
  class="sc-bdVaJa jboVGq"
>
  <ul
    class="sc-bwzfXH fORQjs"
  >
    <li
      class="sc-ifAKCX hQOquv"
      data-testid="one"
      tabindex="0"
    >
      <div
        class="sc-bxivhb eBNJKg"
      >
        one
         
      </div>
    </li>
    <li
      class="sc-ifAKCX jwJMkm"
      data-testid="two"
      tabindex="0"
    >
      <div
        class="sc-bxivhb eBNJKg"
      >
        two
         
      </div>
    </li>
    <li
      class="sc-ifAKCX jwJMkm"
      data-testid="three"
      tabindex="0"
    >
      <div
        class="sc-bxivhb eBNJKg"
      >
        three
         
      </div>
    </li>
  </ul>
  <div
    class="sc-htpNat kEjdxC"
  >
    one
  </div>
</div>
`;
